### ISSUE TITLE
Investigate test failures with "An operation is not legal in the current state."

### ISSUE THREAD

**Issue by wtgodbe** [aspnet/AspNetCore#16817]
Starting with https://github.com/aspnet/AspNetCore/pull/14425, we've been seeing lots of test failures of the following type:

> System.IO.FileLoadException : Could not load file or assembly 'Microsoft.Net.Http.Headers, Version=3.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. An operation is not legal in the current state. (0x80131509)

We think this is related to the large number of Explicit references that we've added to the tests. We should investigate these failures as they may represent an issue with the runtime.

CC @aspnet/build @mmitche

**Comment by dougbu** [comment]
Would like to first determine whether this is about flakiness in our eng\targets\ResolveReferences.targets versus in the .NET SDK. To that end, the initial task here might be to create a project with lots and lots of explicit `<ProjectReference>`s and `<PackageReference>`s but none of the AspNetCore `<Reference>` infrastructure, then build it over and over.

**Comment by mkArtakMSFT** [comment]
@wtgodbe this should move to a proper milestone. I think this milestone is already locked down, so we should close it.

**Comment by wtgodbe** [comment]
We think this will be fixed by https://github.com/aspnet/AspNetCore/pull/17311 - if so I'll close when that's merged

**Comment by dougbu** [comment]
Close once merge into 'release/3.1' is done

**Comment by wtgodbe** [comment]
We stopped seeing these after https://github.com/aspnet/AspNetCore/pull/17311, closing
